src/tike/ptycho/ptycho.py

```diff
@@ -192,7 +192,25 @@ def reconstruct(
                     break
                 cost = result['cost']
 
+            result['psi'], result['probe'] = _rescale_obj_probe(
+                operator, data, result['psi'], result['scan'], result['probe'])
+
         return {k: operator.asnumpy(v) for k, v in result.items()}
     else:
         raise ValueError(
             "The '{}' algorithm is not an available.".format(algorithm))
+
+
+def _rescale_obj_probe(operator, data, psi, scan, probe):
+    """Keep the object amplitude around 1 by scaling by a constant"""
+    intensity = operator._compute_intensity(data, psi, scan, probe)
+
+    rescale = (np.linalg.norm(np.ravel(np.sqrt(intensity))) /
+               np.linalg.norm(np.ravel(np.sqrt(data))))
```

Comment:
Yes, I tried the method you proposed in #47. If the objective is to keep `np.mean(np.abs(psi))` near `1`, then it doesn't work. Instead the amplitude of the object slowly creeps toward zero and the probe amplitude moves to infinity.